apps/institution-service/src/controllers/institutionController.ts

```diff
@@ -97,3 +98,98 @@ export const updateInstitution = async (req: Request, res: Response) => {
       .json({ error: "An error occurred while updating the institution" });
   }
 };
+
+interface AggregatorIntegrator {
+  aggregator_institution_id: string;
+  supports_oauth: boolean;
+  supports_identification: boolean;
+  supports_verification: boolean;
+  supports_aggregation: boolean;
+  supports_history: boolean;
+  isActive: boolean;
+  aggregator: {
+    name: string;
+    id: number;
+    displayName: string | null;
+    logo: string | null;
+  };
+}
+
+interface InstitutionDetail {
+  id: string;
+  name: string;
+  keywords: string[];
+  logo: string;
+  url: string;
+  is_test_bank: boolean;
+  routing_numbers: string[];
+  createdAt: string;
+  updatedAt: string;
+  aggregatorIntegrators: AggregatorIntegrator[];
+}
+export interface PaginatedInstitutionsResponse {
+  currentPage: number;
+  pageSize: number;
+  totalRecords: number;
+  totalPages: number;
+  institutions: InstitutionDetail[];
+}
+
+interface PaginationOptions {
+  page: number;
+  limit: number;
+  offset: number;
+}
+
+const getPaginationOptions = (req: Request): PaginationOptions => {
```

Comment:
I'd write tests for this function as well. At least to check the default behavior